rclpy/test/test_init_shutdown.py

```diff
@@ -74,15 +74,6 @@ def test_double_init():
         rclpy.shutdown(context=context)
 
 
-def test_double_shutdown():
-    context = rclpy.context.Context()
-    rclpy.init(context=context)
-    assert context.ok()
-    rclpy.shutdown(context=context)
-    with pytest.raises(RuntimeError):
```

Comment:
Do we need to remove this test?

I think we can keep this test, but expects `RCLError` exception here to catch from 2nd `rcl_shutdown` (internally `RCL_RET_ALREADY_SHUTDOWN` returns in rcl) ?

Reply:
Thanks for your comments.

> I think we can keep this test, but expects RCLError exception here to catch from 2nd rcl_shutdown (internally RCL_RET_ALREADY_SHUTDOWN returns in rcl) ?

With the current fix, if the context is not in g_contexts, rcl_shutdown will not be called.  

According to the issue described in #1352, it is indeed possible for two threads to call rcl_shutdown on the same context. However, one call is from rclpy::shutdown() and the other from Context::shutdown(). In this situation, we do not want to see an exception being thrown.

The test calls Context::shutdown() twice, and it is reasonable to expect an exception on the second call. Therefore, I am considering setting a flag variable in the Context class. If Context::shutdown() has been successfully called, this variable will be set. If Context::shutdown() is called again, it will check if the flag has been set and, if so, will throw an exception CONTEXT_ALREADY_SHUTDOWN (not from RCL). What do you think?

Reply:
> Therefore, I am considering setting a flag variable in the Context class. If Context::shutdown() has been successfully called, this variable will be set. If Context::shutdown() is called again, it will check if the flag has been set and, if so, will throw an exception CONTEXT_ALREADY_SHUTDOWN (not from RCL). What do you think?

Please review https://github.com/ros2/rclpy/pull/1353/commits/0eb7ee57731d0b2dbf44288493290e5049b55fc0